--- a/piracer/gamepads.py	2023-11-22 17:51:33.555548627 +0100
+++ b/piracer/gamepads.py	2023-11-22 17:52:04.255099906 +0100
@@ -20,10 +20,10 @@
         self.button_l2: Optional[float]
         self.button_r1: Optional[float]
         self.button_r2: Optional[float]
-        self.button_x: Optional[bool]
-        self.button_a: Optional[bool]
-        self.button_b: Optional[bool]
-        self.button_y: Optional[bool]
+        self.button_x = 0
+        self.button_a = 0
+        self.button_b = 0
+        self.button_y = 0
         self.button_select: Optional[bool]
         self.button_start: Optional[bool]
         self.button_home: Optional[bool]
@@ -173,41 +173,22 @@
             self.gamepad_input.analog_stick_left.x = axis_val
         elif axis_number == 1:
             self.gamepad_input.analog_stick_left.y = -axis_val
-        elif button_number == 10:
-            self.gamepad_input.analog_stick_left.z = button_state
-
         elif axis_number == 2:
             self.gamepad_input.analog_stick_right.x = axis_val
         elif axis_number == 3:
             self.gamepad_input.analog_stick_right.y = -axis_val
-        elif button_number == 11:
-            self.gamepad_input.analog_stick_right.z = button_state
-
-        elif button_number == 4:
-            self.gamepad_input.button_l1 = button_state
-        elif button_number == 6:
-            self.gamepad_input.button_l2 = button_state
-
-        elif button_number == 5:
-            self.gamepad_input.button_r1 = button_state
-        elif button_number == 7:
-            self.gamepad_input.button_r2 = button_state
-
-        elif button_number == 2:
+        # Buttons
+        elif button_number  == 0:
             self.gamepad_input.button_a = button_state
-        elif button_number == 1:
+        elif button_number  == 1:
             self.gamepad_input.button_b = button_state
-        elif button_number == 3:
+        elif button_number  == 2:
+            pass
+        elif button_number  == 3:
             self.gamepad_input.button_x = button_state
-        elif button_number == 0:
+        elif button_number  == 4:
             self.gamepad_input.button_y = button_state
-        elif button_number == 1:
-            self.gamepad_input.button_b = button_state
-        elif button_number == 12:
-            self.gamepad_input.button_home = button_state
-        elif button_number == 8:
-            self.gamepad_input.button_select = button_state
-        elif button_number == 9:
-            self.gamepad_input.button_start = button_state
+        elif button_number  == 5:
+            pass
 
         return self.gamepad_input
